### ISSUE TITLE
Concerns regarding the kudosu system in modding v2

### ISSUE THREAD

**Issue by Naxesss** [ppy/osu-web#1698]
Kudosu seems to be completely useless at the moment in modding v2. Although it can allegedly be gained through receiving upvotes on disucssion posts, there is no incentive to actually collect it.

Along with this, the way it's collected doesn't seem to reflect how useful a suggestion was, as the upvote checkpoints are reached through receiving multiple upvotes in a similar fashion to that of the hype system, [which I recently brought up](https://github.com/ppy/osu-web/issues/1686). It also happens to pose the same issues; With it being free and unregulated, anyone can make input regardless of the purpose of the input itself. This would inevitably lead to people upvoting things that are not necessarily helpful to the map, for example memes or a post made by someone they are friends with. There's the possible exploit that a large enough group to trigger one or multiple checkpoints would work together to always upvote each others' posts, giving them more kudosu than others in an unfair way.

On another note, people would be very unlikely to actually look through the discussion and upvote suggestions they think are good due to there being no incentive for that either, especially if the checkpoints for additional kudosu are far in between. Some community members have expressed concerns regarding the navigation of discussion posts in discussions making it difficult to find and sort after what they're looking for, for example sorting posts after users or time posted rather than where on the timeline the timestamp corresponds to, which in turn also somewhat affects this, although being a different topic.

For suggestions on how to improve the collection of kudosu, I'll start with what I would not suggest doing:

- Making people gain kudosu per resolved or applied post, even with a limit to how much kudosu you gain per discussion. This would lead to people posting very short and possibly useless points, in order to maximize kudosu gain while minimizing effort. They could, for example, point out x amount of blankets or suggest changes without explanation, solely to make as many posts of theirs resolved as possible. This is abusable since all discussion posts are encouraged to be resolved before being pushed for ranked.

- Giving more kudosu based on upvotes from anyone. This would lead to an easily exploitable system for the aforementioned reasons. It also essentially makes kudosu as a resource more easily attainable for certain people  but not others, making it seem unfair and lowering the incentive to collect it for the sake of collecting it.

However, what I think would work better is the following:

- Allowing the mapper to give kudosu through upvoting, up to a certain cap per user per discussion or difficulty. This is different from how it is currently in the way that whether a modder gets kudosu or not is up to the mapper. Doing it this way would discourage unhelpful mods, as the mapper would be able to simply not give kudosu, even if the modder made a lot of very minor suggestions that were resolved. This essentially puts the reward system for the modders of the map in the hands of the mapper, allowing them to utilize it however they see fit in order to improve that modding environment, and the map as a result.

It works under the premise that said system is unrelated to any other self-beneficial system. Modding v1, for example, worked with this idea under the star priority system, which rewarded mappers for giving kudosu to any post, regardless of how useful it was, as it would benefit both of them either way.

The other premise is that the mappers have the best interests in mind for how they're giving kudosu. Should they upvote all posts, including very minor or even denied points, this could lead to their discussion being less prosperous and modders preferring to mod their map due to more easily gain kudosu. The cap, along with nominators being able to deny kudosu, would mitigate the damage of this, however.

In conclusion, I don't think the current collection system would be practical, as it would give an incentive for posting something potentially useless to the map, rather than something helpful, along with the mapper being unable to award kudosu by themselves, even though they best judge whether it's helpful themselves. Either way, before implementing a use for kudosu, the way it's earned should probably be addressed first, in order to minimize potential abuse of the way it's currently received, that being through the current upvote checkpoints.
